From 84623cc636431a88b763e05ae2e947d4141ddf29 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Tue, 13 Aug 2019 17:15:26 +0200
Subject: [PATCH] overdrive-split-dt-for-test

- normal dt without frequency scaling
- overdrive dt only frequency scaling
- freqfull with full DVFS

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |  2 ++
 .../fsl-imx8mq-tqma8mq-mba8mx-freqfull.dts    | 30 +++++++++++++++++++
 .../fsl-imx8mq-tqma8mq-mba8mx-overdrive.dts   | 30 +++++++++++++++++++
 .../dts/freescale/fsl-imx8mq-tqma8mq.dtsi     | 13 +-------
 4 files changed, 63 insertions(+), 12 deletions(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq-mba8mx-freqfull.dts
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq-mba8mx-overdrive.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 658c7612d3d5..54ea61598783 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -136,6 +136,8 @@ dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += fsl-imx8mq-ddr3l-arm2.dtb \
 				 fsl-imx8mq-evk-dp.dtb \
 				 fsl-imx8mq-evk-edp.dtb \
 				 fsl-imx8mq-tqma8mq-mba8mx.dtb \
+				 fsl-imx8mq-tqma8mq-mba8mx-overdrive.dtb \
+				 fsl-imx8mq-tqma8mq-mba8mx-freqfull.dtb \
 				 fsl-imx8mq-tqma8mq-mba8mx-cma.dtb \
 				 fsl-imx8mq-tqma8mq-mba8mx-dcss-lvds-tm070jvhg33.dtb \
 				 fsl-imx8mq-tqma8mq-mba8mx-dcss-lvds-tm070jvhg33-en_gpio.dtb \
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq-mba8mx-freqfull.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq-mba8mx-freqfull.dts
new file mode 100644
index 000000000000..eeef824ec3b5
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq-mba8mx-freqfull.dts
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2019 TQ Systems GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8mq-tqma8mq-mba8mx.dts"
+
+&reg_tqma8mx_dvfs {
+	status = "okay";
+};
+
+&A53_0 {
+	operating-points = <
+		/* kHz    uV */
+		1500000 1000000
+		1300000 1000000
+		1000000 900000
+		800000  900000
+	>;
+	dc-supply = <&reg_tqma8mx_dvfs>;
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq-mba8mx-overdrive.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq-mba8mx-overdrive.dts
new file mode 100644
index 000000000000..1966348bc4bb
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq-mba8mx-overdrive.dts
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2019 TQ Systems GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8mq-tqma8mq-mba8mx.dts"
+
+&reg_tqma8mx_overdrive {
+	status = "okay";
+};
+
+&A53_0 {
+	operating-points = <
+		/* kHz    uV */
+		1500000 1000000
+		1300000 1000000
+		1000000 1000000
+		800000  1000000
+	>;
+	arm-supply = <&reg_tqma8mx_overdrive>;
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi
index f4d459cada9f..fd34bb091ddf 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi
@@ -71,7 +71,7 @@
 		regulator-settling-time-us = <50000>;
 		gpios = <&gpio1 6 GPIO_ACTIVE_LOW>;
 		states = <900000 0x1 1000000 0x0>;
-		status = "okay";
+		status = "disabled";
 	};
 
 	reg_tqma8mx_overdrive: tqma8mx-vdd-overdrive {
@@ -94,17 +94,6 @@
 	};
 };
 
-&A53_0 {
-	operating-points = <
-		/* kHz    uV */
-		1500000 1000000
-		1300000 1000000
-		1000000 900000
-		800000  900000
-	>;
-	dc-supply = <&reg_tqma8mx_dvfs>;
-};
-
 &gpu {
 	status = "okay";
 };
